app/javascript/components/enrollment/steps/Contact.js

```diff
@@ -343,56 +334,69 @@ class Contact extends React.Component {
-                    />
-                    <div className="mt-3">
-                      <span className="font-weight-bold">Morning: </span>
-                      <span className="font-weight-light">Between 8:00 and 12:00 in monitoree&apos;s timezone</span>
-                      <br />
-                      <span className="font-weight-bold">Afternoon: </span>
-                      <span className="font-weight-light">Between 12:00 and 16:00 in monitoree&apos;s timezone</span>
-                      <br />
-                      <span className="font-weight-bold">Evening: </span>
-                      <span className="font-weight-light">Between 16:00 and 20:00 in monitoree&apos;s timezone</span>
-                    </div>
-                    <Form.Control.Feedback className="d-block" type="invalid">
-                      {this.state.errors['preferred_contact_time']}
-                    </Form.Control.Feedback>
-                  </Form.Group>
-                )}
+                    )}
+                    onChange={e =>
+                      this.handleChange({
+                        target: { id: 'preferred_contact_time', value: e.value },
+                        currentTarget: { id: 'preferred_contact_time' },
+                      })
+                    }
+                    styles={preferredContactTimeSelectStyling}
+                    theme={theme => bootstrapSelectTheme(theme, 'lg')}
+                  />
+
+                  <div className="mt-3">
+                    <span className="font-weight-bold">Morning: </span>
+                    <span className="font-weight-light">Between 8:00 and 12:00 in monitoree&apos;s timezone</span>
+                    <br />
+                    <span className="font-weight-bold">Afternoon: </span>
+                    <span className="font-weight-light">Between 12:00 and 16:00 in monitoree&apos;s timezone</span>
+                    <br />
+                    <span className="font-weight-bold">Evening: </span>
+                    <span className="font-weight-light">Between 16:00 and 20:00 in monitoree&apos;s timezone</span>
+                  </div>
+                  <Form.Control.Feedback className="d-block" type="invalid">
+                    {this.state.errors['preferred_contact_time']}
+                  </Form.Control.Feedback>
+                  {![null, ''].includes(this.state.current.patient.preferred_contact_time) && (
+                    <React.Fragment>
+                      {[null, ''].includes(this.state.current.patient.preferred_contact_method) &&
+                        this.renderWarningBanner(
+                          'The monitoree will not be sent reminders while they do not have a Preferred Reporting Method selected.',
+                          false,
+                          'warning'
+                        )}
+                      {['Unknown', 'Opt-out'].includes(this.state.current.patient.preferred_contact_method) &&
```

Comment:
Is there a way to clean up the logic here so that you don't need two separate instances of render warning banner?

Reply:
There could be other ways to rewrite this logic slightly differently, but not in a way that significantly reduces the amount of code. The warning banner is only ever rendered once because `preferred_contact_method` can only be one of these values. Let me know if you have any better ideas though

Reply:
hmm I'd have to think about it.  if I think of something better, I can change it in my alternate contact stuff